openff/toolkit/topology/molecule.py

```diff
@@ -3463,6 +3470,21 @@ def assign_partial_charges(
                 f"Expected ToolkitRegistry or ToolkitWrapper. Got  {type(toolkit_registry)}"
             )
 
+    def _normalize_partial_charges(self):
+        """
+        Add offsets to each partial charge to ensure that they sum to the formal charge of the molecule,
+        to the limit of a python float's precision. Modifies the partial charges in-place.
+        """
+        expected_charge = self.total_charge
+
+        current_charge = 0.0 * unit.elementary_charge
+        for pc in self.partial_charges:
+            current_charge += pc
```

Comment:
Suggestion: Could write this using `sum`:
```
current_charge = sum(self.partial_charges, 0.0 * unit.elementary_charge)
```

Reply:
What's wrong with `np.sum(self.partial_charges)`? It doesn't start with the 0 base case so no unit errors.

Reply:
Indeed! My cheminformatics work so rarely uses numpy that I've got a blind spot.

Reply:
This is awesome! I've switched the manual partial charge sums throughout test_toolkits to this one-liner in b71702d.

Reply:
Out of curiosity, what's the problem with numpy?

Reply:
Sorry about that @lilyminium - I saw your message in my email but GitHub didn't show it when I loaded the page, and I figured you had deleted it for some reason. 

I don't have a really strong preference between the two. The additional argument in @adalke's approach is a small overhead, but so is using numpy when a builtin function could work. I suppose the builtin `sum` may be a little more future-proof if we switch to a unit package that doesn't play well with numpy? Though, numpy compatibility seems like the most important thing we'd look for in a units package. Anyway, both are way better than the manual sum that we were using before.

Reply:
I'd probably think of numpy as more transferrable.

Regardless, I've just realised you can do `self.partial_charges.sum()`, which might be the best solution overall. I think that both behaviours are compatible with pint as well, which IIRC was one of the major contenders for unit replacement?

Reply:
> Out of curiosity, what's the problem with numpy?

No so much a problem, rather that I rarely need it in my code. Take a look at https://github.com/rdkit/mmpdb/ - not an array of numerical data in sight, nor use of anything else in NumPy. Lack of experience with NumPy means I have to think harder to consider NumPy solutions over built-in Python ones.